src/cache-worker.ts

```diff
@@ -101,57 +150,43 @@ export class CacheWorker {
     return this.grahQLClient.request(query, variables)
   }
 
-  private async handleError(
-    updateCachePromise: Promise<GraphQLResponse>,
-    currentKeys: string[]
-  ) {
-    await updateCachePromise
-      .then(async (graphQLResponse) => {
-        if (graphQLResponse.errors) {
-          await this.retry(currentKeys)
-        }
-        this.fillLogs(graphQLResponse)
-      })
-      .catch(async (error: GraphQLError) => {
-        await this.retry(currentKeys)
-        this.fillLogs(error)
-      })
+  private bisect(task: Task) {
+    const splittedKeys = splitEvery(task.keys.length / 2, task.keys)
+    splittedKeys.forEach((keys) => {
+      const task = { keys, numberOfRetries: 0 }
+      this.tasks.push(task)
+    })
```

Comment:
```suggestion
    splitEvery(task.keys.length / 2, task.keys).forEach((keys) => {
      this.tasks.push({ keys, numberOfRetries: 0 })
    })
```

I would do it like this...